### PR TITLE
CF/BF - Update configurator for API 1.33.0.

### PR DESCRIPTION
**Pull request by hydra**
PR goals:
- [x] Add support for API 1.33.0.
- [x] Use specific commands instead of MSP_MISC and MSP_BF_CONFIG.
- [x] Drop support for all versions of CF < 2.0 and all version of BF < 3.0
- [ ] Port and update the 'Power and Battery' tab from CF configurator v1.x
- [x] Add stick configuration (min/max) to RX tab (from CF configurator v1.x)
- [x] Move 3D throttle deadband to RX tab (from CF configurator v1.x)
- [ ] ~~Move RX configuration from Config tab to RX tab (from CF configurator v1.x)~~
- [ ] ~~Move RSSI configuration from Config tab to RX tab.~~
- [x] Delete any old code found along the way.
- [x] Support ILAP transponders via updated MSP_TRANSPONDER_CONFIG (See https://github.com/cleanflight/cleanflight/pull/2580 and https://github.com/cleanflight/cleanflight/pull/2690) - Additional PR required to add ARCITIMER transponder configuration.

Testing is currently being done with CF 2.0.0 master.

Although some items are marked as 'from CF configurator v1.x' not that much actual js code is being used due to recent changes, however the html/css/messages are mainly work ok so the porting is manual.

### CODE REVIEW

**Comment by hydra** [comment]
Currently the most tabs are working OK.

- [ ] motors tab needs fixing.

File: js/fc.js [hunk 1]
```diff
@@ -182,31 +225,50 @@ var FC = {
         FC_CONFIG = {
             loopTime: 0
         };
-
-        MISC = {
+
```
Comment (mikeller) [inline js/fc.js]: Space.

File: js/fc.js [hunk 2]
```diff
@@ -182,31 +225,50 @@ var FC = {
...
+//            vbatwarningcellvoltage: 0,
+//            batterymetertype:       1, // 1=ADC, 2=ESC
+//        };
+        
+        MOTOR_3D_CONFIG = {
             deadband3d_low:         0,
             deadband3d_high:        0,
-            neutral3d:              0,
-            deadband3d_throttle:    0
+            neutral:              0,
```
Comment (mikeller) [inline js/fc.js]: Space.

File: js/fc.js [hunk 3]
```diff
@@ -296,9 +359,9 @@ var FC = {
 
         RX_CONFIG = {
             serialrx_provider:      0,
-            maxcheck:               0,
-            midrc:                  0,
-            mincheck:               0,
+            stick_max:               0,
+            stick_center:                  0,
```
Comment (mikeller) [inline js/fc.js]: Space.

File: js/msp.js [hunk 4]
```diff
@@ -148,6 +148,9 @@ var MSP = {
         this.listeners = [];  
     },
     send_message: function (code, data, callback_sent, callback_msp, callback_onerror) {
+        if (code === undefined) {
+            debugger;
```
Comment (mikeller) [inline js/msp.js]: Intentionally throwing an exception?

File: tabs/onboard_logging.js [hunk 5]
```diff
@@ -110,7 +110,7 @@ TABS.onboard_logging.initialize = function (callback) {
              * The best we can do on those targets is check the BLACKBOX feature bit to identify support for Blackbox instead.
              */
             if (BLACKBOX.supported || DATAFLASH.supported 
-                    || semver.gte(CONFIG.flightControllerVersion, "1.5.0") && semver.lte(CONFIG.flightControllerVersion, "1.10.0") && BF_CONFIG.features.isEnabled('BLACKBOX')) {
+                    || semver.gte(CONFIG.flightControllerVersion, "1.5.0") && semver.lte(CONFIG.flightControllerVersion, "1.10.0") && FEATURE_CONFIG.features.isEnabled('BLACKBOX')) {
```
Comment (mikeller) [inline tabs/onboard_logging.js]: Should this be checking `apiVersion` instead?

File: js/fc.js (hunk 1 above)
Comment (hydra) [inline js/fc.js]: MISC will be deleted soon 😄

File: js/fc.js (hunk 3 above)
Comment (hydra) [inline js/fc.js]: there will be many spaces, i did LOTS of search and replacing today.  I'll go over it again after it's functional, no point aligning things at this stage.

File: js/msp.js (hunk 4 above)
Comment (hydra) [inline js/msp.js]: yes, to catch the stacktrace, will be removed from production code.

File: tabs/onboard_logging.js (hunk 5 above)
Comment (hydra) [inline tabs/onboard_logging.js]: probably.  I'll be stripping a out all the old conditional code that's not currently getting in the way in later commits.

**Comment by hydra** [comment]
current status, fully working but lacks the battery/power tab to configure voltage meters.

**Comment by hydra** [comment]
rebased against master.

I'm going to de-scope this PR. what's there is fine for a release, we can do the remainder of the tasks in this PR later.

**Comment by hydra** [comment]
currently the code is working, there's a few minor cleanups to do which I will do now before merging to master.

**Comment by hydra** [comment]
only thing really missing right now is current and voltage meter configuration.  what's done so far needs to be released - will follow up with another PR for additional work/features.

**Pull request merged**